app/util/k8s/terminate_cluster.py

```diff
@@ -981,7 +981,11 @@ def delete_expired_tf_state_s3_buckets():
         for bucket in response["Buckets"]:
             if bucket_name_template in bucket["Name"]:
                 created_date = bucket["CreationDate"]
-                tags = s3_client.get_bucket_tagging(Bucket=bucket["Name"])["TagSet"]
+                try:
+                    tags = s3_client.get_bucket_tagging(Bucket=bucket["Name"])["TagSet"]
+                except s3_client.exceptions.NoSuchTagSet:
+                    logging.warning(f"S3 bucket {bucket['Name']} does not have any tags.")
+                    continue
```

Comment:
Let's make a re-raise exception here with a clear error message containing the bucket name.
It would be better to fail the plan and have an engineer to take a look and decide what to do with a bucket, rather just skipping.